hw/ip/otbn/dv/otbnsim/sim/stats.py

```diff
@@ -0,0 +1,302 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from collections import Counter
+from typing import Dict, List, Optional, Tuple
+
+from elftools.dwarf.descriptions import describe_form_class
+from elftools.dwarf.dwarfinfo import DWARFInfo
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import SymbolTableSection
+from tabulate import tabulate
+
+from .insn import JAL, JALR, LOOP, LOOPI
+from .isa import OTBNInsn
+from .state import OTBNState
+from .trace import Trace
+
+
+class ExecutionStats:
+    def __init__(self) -> None:
+        self.stall_count = 0
+        self.insn_count = 0
+        self.instruction_histo = Counter()
+        self.func_calls = [] # type: List[Dict[str, str]]
+        self.loops = [] # type: List[Dict[str, str]]
```

Comment:
Nit: I think you'll see flake8 warnings if there isn't a double space before the hash.